### PR TITLE
Fix comment bugs

### PR DESCRIPTION
**Pull request by srfarley**
- [x] @sfeng88 @rygim @jharwig 
- [x] @joeferner
- [x] @EvanOxfeld @joeybrk372
- [x] @mwizeman @dsingley

Fixes the following bugs:
* Edit button is not displayed for author's own comments.
* Deleting a comment leaves the Delete button with spinner displayed indefinitely.
* Comment metadata for other users is not visible when clicking on the "i".

Other improvements:
* Remove duplicate code from set/delete vertex/edge routes, and move into `ACLProvider`.
* Support and enforce `/vertex/comment` and `/edge/comment` URLs for delete.

### CODE REVIEW

File: web/war/src/main/webapp/js/data/web-worker/services/storeHelper.js [hunk 1]
```diff
@@ -57,6 +57,11 @@ define(['../util/store'], function(store) {
                         })
                 }
             };
+        },
+
+        propertyUrl: function(elementType, property) {
+            return '/' + elementType + '/' +
+                (property.name === 'http://visallo.org/comment#entry' ? 'comment' : 'property');
```
Comment (EvanOxfeld) [inline web/war/src/main/webapp/js/data/web-worker/services/storeHelper.js]: I recommend making `propertyUrl` an internal function and exposing something like `vertexPropertyUrl` and `edgePropertyUrl` on the public `storeHelper` API. Reducing the number of arguments to one makes the function more readable and easier to use. Also, it's one fewer instance where the string `vertex` or `edge` is passed around.

Comment (srfarley) [inline web/war/src/main/webapp/js/data/web-worker/services/storeHelper.js]: @EvanOxfeld Done.

**Comment by sfeng88** [comment]
@jharwig backend looks good to me, can you check the front end please?

**Comment by jharwig** [comment]
:+1:

**Comment by dsingley** [comment]
Rename `ChangeAllAclProvider` and `ChangeNoneAclProvider` to `AllowAll` and `AllowNone`?

**Comment by srfarley** [comment]
@dsingley Done.

**Pull request merged**